distributed/protocol/compression.py

```diff
@@ -39,56 +41,35 @@
 with suppress(ImportError):
     import snappy
 
-    def _fixed_snappy_decompress(data):
-        # snappy.decompress() doesn't accept memoryviews
-        if isinstance(data, (memoryview, bytearray)):
-            data = bytes(data)
-        return snappy.decompress(data)
```

Comment:
Support for the Python Buffer Protocol was added in [0.5.3]( https://github.com/andrix/python-snappy/releases/tag/0.5.3 ) with PR ( https://github.com/andrix/python-snappy/pull/72 ). This version was released Jul 2018 so should be old enough to rely upon. With Python Buffer Protocol support, there is no longer a need to copy too `bytes` here and we can just hand `data` to `snappy.decompress`.

Reply:
+1